### ISSUE TITLE
Content Fetching and Post Processing Architecture

### ISSUE THREAD

**Issue by 69mb** [ushahidi/SwiftRiver#23]
Droplet acquisition and post processing needs to be built out better with an emphasis on scalability of the workflow from the get go.

**Comment by 69mb** [comment]
Currently ironing this out...

**Comment by dkobia** [comment]
Quick question @69mb / @ekala -- Down the road would it be possible to have gearman worker processes that deal with droplets processing farmed out to other servers? Trying to think ahead.

**Comment by 69mb** [comment]
Yes, that is the idea. With the scale of processing that will be required in a hosted setup, jobs will have to be outsourced to highly specialized systems outside the webserver. For instance, I suspect the rss crawler will look something like [this](http://3.bp.blogspot.com/-C99QChtEdoY/TsGF6ZUCHEI/AAAAAAAAACU/23tXAxtAHe4/s1600/SpiderDuck-Architecture.jpg) int the future...

Removing gear man however in a commit I will be pushing to the remote soon for the following reasons:
1. It is not transparent enough. There is absolutely no log / way to tell what it is currently doing.
2. Found that at the moment, without farming out work to external specialized workers, it is simply adding an unecessary level of complexity in situations that can be handled by a simple fork or where no parallelism is required at all.

Basically when the time comes, we will add integrate a queueing system but at the moment, let us explore the limits of simple crawling.

**Comment by ekala** [comment]
Some notes:
1. Gearman has an "Administrative Protocol" which you can use for retrieving current status of workers, registered functions etc http://gearman.org/?id=protocol Only downside is that it works over "telnet" or "nc". There's also a nagios plugin for automated monitoring and notifications.
2. Yes. Gearman's logging is not very good (probably non-existent)
